service-statistics-pxweb/src/main/java/fi/nls/oskari/control/statistics/plugins/pxweb/parser/PxwebIndicatorsParser.java

```diff
@@ -82,13 +82,11 @@ public PxwebIndicatorsParser(PxwebConfig config) {
                         ind.setId(indicatorId);
                         indicatorMap.put(indicatorId, ind);
                         indicators.add(ind);
-                    }
-                    ind.addName(lang, item.text);
-                    if (ind.getDataModel() == null) {
```

Comment:
Replace null check with:
```
if (ind.getDataModel().getDimensions().isEmpty()) {
     ind.setDataModel(getModel(table));
}
```

If the null check is removed entirely the indicator params will have the last language that was supported by the API. So for example in paikkatietoikkuna this would mean that the Helsinki datasource would have indicator names in finnish but the selectors/parameters in english or swedish. Since we don't have the selector merging implemented yet the current code tries to have the selectors with the primary language of the geoportal. This leads to indicator name being localized but the selectors always in finnish on paikkatietoikkuna.

Reply:
Nevermind, this is ok as the datamodel is only created for new indicators.